README.md

```diff
@@ -139,7 +139,126 @@ import utc from '@momentjs/moment/timezones/utc/'; // Note trailing slash
+
+Conditional mapping is an extension of the `"exports"` and `"default"` features that allows defining different mappings between different environments, for example having a different module path between Node.js and the browser.
+
+Conditional mappings are defined as _objects_ in the target slot for both `"exports"` and the `"default"` field.
+
+The object has keys which are _condition names_, and values which correspond to the mapping target.
+
+Condition names are matched in a priority order. In Node.js the following conditions apply in priority order:
+
+1. `"require"`: Indicates we are resolving from a CommonJS importer.
+2. `"node"`: Indicates we are in a Node.js environment.
+
+> Note: Using a "require" condition opens up the dual specifier hazard in Node.js where a package can have different instances between CJS and ESM importers. There is an argument that this condition is an opt-in behaviour to the hazard which is less risky than the main concerns of the hazard which were non-intentional cases. It is still not clear if this condition will get consensus, and it may still be removed.
+
+> Node: It could also be worthwhile considering object order over priority order. This would shift the source of truth of the entry priority from the resolver to the package author, which could be beneficial, but should be discussed further.
+
+If no condition is matched, the package fallback applies. If the matched condition target is invalid, the fallback will still not apply, although individual condition targets can themselves use array fallbacks.
+
+Other resolvers are free to define their own conditions to match. Eg it is expected that users will use a `"browser"` condition name for browser mappings.
+
+#### Example
+
+Taking the previous moment example we can provide browser / Node.js mappings for some modules with the following:
+
+```js
+{
+  "name": "@momentjs/moment",
+  "version": "0.0.0",
+  "type": "module",
+  "main": "./dist/index.js",
+  "default": {
+    "node": "./dist/index.js",
+    "browser": "./dist/index-browser.js"
+  },
+  "exports": {
+    "./": {
+      "node": "./src/util/",
+      "browser": "./src/util-browser/"
+    },
+    "./timezones/": "./data/timezones/",
+    "./timezones/utc": "./data/timezones/utc/index.mjs",
+    "./core-polyfill": {
+      "node": ["std:core-module", "./core-polyfill"],
+      "browser": "./core-polyfill-browser"
+    }
+  }
+}
+```
+
+Note we are able to share some exports with both Node.js and the browser and others split between the environments.
+
+#### Dual package example
+
+For an example of a package that wants to support legacy Node.js, `require()` and `import` (noting that this is an opt-in to the instancing hazard, and pending consensus), we can use the `"require"` condition which will always beat the `"node"` condition as it is a more specific condition:
+
+```js
+{
+  "type": "module"
+  "main": "./index-legacy.cjs",
+  "default": [{
+    "require": "./index.cjs"
+  }, "./index.js"],
```

Comment:
`require` will match exactly only in `--experimental-modules` being unflagged. So the default being ignored is exactly anything that is not "Node.js supporting modules". index-legacy.cjs and index.cjs can be the same, but this example is just showing that differentiation can be made. Perhaps it would be simpler to combine them for this example though.